### PR TITLE
Tree: reorganize cursor tests

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Cleanup cursor test organization.  Adds more test coverage more mapTree and forest cursors.

### CODE REVIEW

File: packages/dds/tree/src/test/cursorTestSuite.ts [hunk 1]
```diff
@@ -117,18 +118,29 @@ export const cursorTestCases: [string, JsonableTree][] = [
...
+    dataFromCursor: (cursor: ITreeCursor) => T,
     testRooted = true,
 ): void {
+    function dataFromJsonableTree(data: JsonableTree): T {
+        // Use text cursor to provide input data
+        return dataFromCursor(singleTextCursor(data));
+    }
+    function factory(data: JsonableTree): ITreeCursor {
+        return cursorFactory(dataFromJsonableTree(data));
+    }
```
Comment (jenn-le) [inline packages/dds/tree/src/test/cursorTestSuite.ts:133]: Nit: I personally like white space. Feel free to ignore this comment though.

```suggestion
    }
    
    function factory(data: JsonableTree): ITreeCursor {
        return cursorFactory(dataFromJsonableTree(data));
    }
    
```

File: packages/dds/tree/src/test/cursorTestSuite.ts [hunk 2]
```diff
@@ -117,18 +118,29 @@ export const cursorTestCases: [string, JsonableTree][] = [
...
     describe(`${cursorName} cursor implementation`, () => {
         describe("extract roundtrip", () => {
             for (const [name, data] of cursorTestCases) {
                 it(`${name}: ${JSON.stringify(data)}`, () => {
-                    const cursor = factory(data);
-                    const clone = dataFromCursor(cursor);
+                    const convertedData = dataFromJsonableTree(data);
+                    const cursor = cursorFactory(convertedData);
+                    const convertedClone = dataFromCursor(cursor);
+                    // This assumes `T` is works with deepEqual.
```
Comment (jenn-le) [inline packages/dds/tree/src/test/cursorTestSuite.ts:141]:
```suggestion
                    // This assumes `T` works with deepEqual.
```

File: packages/dds/tree/src/test/feature-libraries/objectForest.spec.ts [hunk 3]
```diff
@@ -90,6 +93,24 @@ function testForest(
             }
         });
 
+        it("cursor use", () => {
+            //
```
Comment (jenn-le) [inline packages/dds/tree/src/test/feature-libraries/objectForest.spec.ts:97]: Is there supposed to be a comment here?

**Review by jenn-le — APPROVED** [review]

File: packages/dds/tree/src/test/cursorTestSuite.ts (hunk 1 above)
Comment (CraigMacomber) [reply packages/dds/tree/src/test/cursorTestSuite.ts:133]: Your suggestion has training white space on the blank line.
```suggestion
    }

    function factory(data: JsonableTree): ITreeCursor {
        return cursorFactory(dataFromJsonableTree(data));
    }
```

File: packages/dds/tree/src/test/feature-libraries/objectForest.spec.ts [hunk 4]
```diff
@@ -90,6 +93,24 @@ function testForest(
...
+            const delta: Delta.Root = new Map([[rootFieldKeySymbol, [setValue]]]);
+            forest.applyDelta(delta);
+
+            const reader = forest.allocateCursor();
+            moveToDetachedField(forest, reader);
+            assert(reader.firstNode());
+
+            assert.equal(reader.value, 2);
+        });
+
```
Comment (CraigMacomber) [inline packages/dds/tree/src/test/feature-libraries/objectForest.spec.ts:113]:
```suggestion
```

Comment (CraigMacomber) [reply packages/dds/tree/src/test/feature-libraries/objectForest.spec.ts:113]: THat was a test I was working on for something else when I started that. Shouldn't be in this PR.

File: packages/dds/tree/src/test/cursorTestSuite.ts (hunk 1 above)
Comment (jenn-le) [reply packages/dds/tree/src/test/cursorTestSuite.ts]: Oh whoops, nice catch.

File: packages/dds/tree/src/test/feature-libraries/objectForest.spec.ts (hunk 3 above)
Comment (CraigMacomber) [reply packages/dds/tree/src/test/feature-libraries/objectForest.spec.ts]: this wasn't supposed to be in this PR: removed.

**Pull request merged**